testgen/generators.go

```diff
@@ -186,44 +187,100 @@ var EthGetCode = MethodTests{
 	[]Test{
 		{
 			"get-code",
-			"gets code for 0xaa",
+			"requests code of an existing contract",
 			func(ctx context.Context, t *T) error {
-				addr := common.Address{0xaa}
 				var got hexutil.Bytes
-				err := t.rpc.CallContext(ctx, &got, "eth_getCode", addr, "latest")
+				err := t.rpc.CallContext(ctx, &got, "eth_getCode", emitContract, "latest")
 				if err != nil {
 					return err
 				}
-				state, _ := t.chain.State()
-				want := state.GetCode(addr)
+				want := t.chain.state[emitContract].Code
 				if !bytes.Equal(got, want) {
 					return fmt.Errorf("unexpected code (got: %s, want %s)", got, want)
 				}
 				return nil
 			},
 		},
+		{
+			"get-code-unknown-account",
+			"requests code of a non-existent account",
+			func(ctx context.Context, t *T) error {
+				var got hexutil.Bytes
+				err := t.rpc.CallContext(ctx, &got, "eth_getCode", nonAccount, "latest")
+				if err != nil {
+					return err
+				}
+				if len(got) > 0 {
+					return fmt.Errorf("account %v has non-empty code", nonAccount)
+				}
+				return nil
+			},
+		},
 	},
 }
 
 // EthGetStorage stores a list of all tests against the method.
 var EthGetStorage = MethodTests{
-	"eth_getStorage",
+	"eth_getStorageAt",
 	[]Test{
 		{
 			"get-storage",
-			"gets storage for 0xaa",
+			"gets storage of a contract",
 			func(ctx context.Context, t *T) error {
-				addr := common.Address{0xaa}
-				key := common.Hash{0x01}
+				addr := emitContract
+				key := common.Hash{}
 				got, err := t.eth.StorageAt(ctx, addr, key, nil)
 				if err != nil {
 					return err
 				}
-				state, _ := t.chain.State()
-				want := state.GetState(addr, key)
-				if !bytes.Equal(got, want.Bytes()) {
+				want := t.chain.Storage(addr, key)
+				if !bytes.Equal(got, want) {
 					return fmt.Errorf("unexpected storage value (got: %s, want %s)", got, want)
 				}
+				// Check for any non-zero byte in the value.
+				// If it's all-zero, the slot doesn't really exist, indicating a problem with the test itself.
+				nz := slices.ContainsFunc(got, func(b byte) bool { return b != 0 })
+				if !nz {
+					return fmt.Errorf("requested storage slot is zero")
+				}
+				return nil
+			},
+		},
+		{
+			"get-storage-unknown-account",
+			"gets storage of a non-existent account",
+			func(ctx context.Context, t *T) error {
+				key := common.Hash{1}
+				got, err := t.eth.StorageAt(ctx, nonAccount, key, nil)
+				if err != nil {
+					return err
+				}
+				nz := slices.ContainsFunc(got, func(b byte) bool { return b != 0 })
```

Comment:
why not just allocate empty hash and check equality?

Reply:
I did the check in this way because the return value is `[]byte`. The way it's done, the length of `got` is not important.